tests/s2_inference/test_generic_clip_model.py

```diff
@@ -0,0 +1,387 @@
+import numpy as np
+
+from marqo.errors import IndexNotFoundError
+from marqo.s2_inference.errors import InvalidModelPropertiesError, UnknownModelError, ModelLoadError
+from marqo.tensor_search import tensor_search
+from marqo.s2_inference.processing.custom_clip_utils import download_pretrained_from_url
+from marqo.s2_inference.s2_inference import clear_loaded_models
+
+from marqo.s2_inference.s2_inference import (
+    available_models,
+    vectorise,
+    _validate_model_properties,
+    _update_available_models
+)
+
+from tests.marqo_test import MarqoTestCase
+
+
+class TestGenericModelSupport(MarqoTestCase):
+
+    def setUp(self):
+        self.index_name_1 = "my-test-index-1"
+        self.index_name_2 = "my-test-index-2"
+        try:
+            tensor_search.delete_index(config=self.config, index_name=self.index_name_1)
+        except IndexNotFoundError as e:
+            pass
+
+
+    def tearDown(self) -> None:
+        try:
+            tensor_search.delete_index(config=self.config, index_name=self.index_name_1)
+        except IndexNotFoundError as e:
+            pass
+        try:
+            tensor_search.delete_index(config=self.config, index_name=self.index_name_2)
+        except IndexNotFoundError as e:
+            pass
+        clear_loaded_models()
+
+
+    def test_create_index_and_add_documents_with_generic_open_clip_model_properties_url(self):
+        """index should get created with custom model_properties
+        """
+        # Step1 - Create Index
+        model_name = 'test-model-1'
+        model_properties = {"name": "ViT-B-32-quickgelu",
+                            "dimensions": 512,
+                            "url": "https://github.com/mlfoundations/open_clip/releases/download/v0.2-weights/vit_b_32-quickgelu-laion400m_avg-8a00ab3c.pt",
+                            "type": "open_clip",
+                            }
+        tensor_search.create_vector_index(
+            index_name=self.index_name_1, config=self.config,
+            index_settings={
+                "index_defaults": {
+                    'model': model_name,
+                    'model_properties': model_properties
+                }
+            }
+        )
+        # Step2 - Add documents
+        docs = [
+            {
+                "_id": "123",
+                "title 1": "content 1",
+                "desc 2": "content 2. blah blah blah"
+            }]
+
+        auto_refresh = True
+        tensor_search.add_documents(config=self.config, index_name=self.index_name_1, docs=docs, auto_refresh=auto_refresh)
+
+        # Step3 - Search
+        results = tensor_search.search(config=self.config, index_name=self.index_name_1, text = "test-test")
```

Comment:
How about making a few more assertions?
- That the intended document was retrieved?
- And that an extra doc was added to the index?
 - And that the score was reasonable?